File: src/io/innofang/strategy/example/java/Wang.java
package io.innofang.strategy.example.java;

/**
 * Created by Inno Fang on 2017/3/3.
 * <p>
 * Context 类
 */
public class Wang {

    private GoToStrategy goToStrategy;

    public void setGoToStrategy(GoToStrategy goToStrategy) {
        this.goToStrategy = goToStrategy;
    }

    public void take() {
        goToStrategy.transportation();
    }
}

File: src/io/innofang/strategy/example/kotlin/Wang.kt
package io.innofang.strategy.example.kotlin


/**
 * Created by Inno Fang on 2017/10/24.
 *
 *
 * Context 类
 */
class Wang {

    var goToStrategy: GoToStrategy? = null

    fun take() {
        goToStrategy?.transportation()
    }
}